## Estadistica univariada 

#los datos estructurados que estan en el excel son los siguientes y los
voy a lista
vendedor del producto
fecha límite de envio
precio del producto
valro del flete
código postal vendedor
ciudad vendedor
departamento del vendedor
nombre categoría de producto
cantidad fotos del producto
peso en g del prodcuto
longitud del producto
altura en cm del prodcuto
ancho del prodcuto


In [2]:
#improtar las librerías necesarias según el análisis que se vaya a realizar
#libreria para comando de sistema
import os  #os es un módulo de python 
import pandas as pd #pandas es una librerpia de python especializada en manejo y análisis de estructuras de datos


In [3]:
#cargar los datos en CSV
datos=pd.read_csv("Ordenes_productos_C1_M2.csv", sep=";", encoding="latin-1")

In [4]:
##lectura de datos
#ES IMPORTANTE TENER EN CUENTA LA VISUALIZACION DE LOS DATOS PORQUE
#ADEMÁS DE CONTEXTO se puede entender mucho mejor de qué se está hablando y qué tipos de datos
#se han cargado desde la máquina


In [5]:
#visualizacion de lso datos 
datos.head()

orden_id order_item_id producto_id vendedor_id fecha_envio_limite  \
0  107500PO59A             A        PO59      VE5389    7/04/2018 18:12   
1   37493PS22B             B        PS22      VE1558    20/10/2017 9:07   
2   28050PK20B             B        PK20      VE9159    17/08/2017 8:15   
3   52187PA10A             A        PA10      VE3159   23/09/2017 23:27   
4   84639PR12A             A        PR12      VE5090    7/01/2018 11:50   

   precio  valor_flete  codigo_postal_vendedor ciudad_vendedor  \
0  271.86        30.72                   52435         Mallama   
1  115.73         4.68                   52203           Colon   
2  432.99        82.70                   66001         Pereira   
3  108.38        35.39                   52435         Mallama   
4   51.50        11.10                   73001          Ibague   

  departamento_vendedor nombre_categoria_producto  longitud_nombre_producto  \
0                Nariño    Productos ecoamigables                         6   
1                Nariño                Carnicería                        10   
2             Risaralda                  Deportes                        25   
3                Nariño         Electrodomésticos                        10   
4                Tolima         Frutas y verduras                        23   

   longitud_descripcion_producto  cantidad_fotos_producto  peso_g_producto  \
0                              7                       27             2486   
1                             31                       20              256   
2                              5                        4             5270   
3                              1                        6              734   
4                             16                       35              884   

   longitud_cm_producto  altura_cm_producto  ancho_cm_producto  
0                    17                  11                 14  
1                    43                   2                 21  
2                     9                  27                 29  
3                    46                  48                 22  
4                    45                  26                 18

los tipos de datos nos van a ayudar a entender qué tanálisis se puede realizar con cada variable para su correcta interpretación

In [6]:
#revisar los tipos de datos en el archivo
datos.dtypes # me ayuda a entender qué tipo de operaciones se puedne hacer con cada dato

orden_id                          object
order_item_id                     object
producto_id                       object
vendedor_id                       object
fecha_envio_limite                object
precio                           float64
valor_flete                      float64
codigo_postal_vendedor             int64
ciudad_vendedor                   object
departamento_vendedor             object
nombre_categoria_producto         object
longitud_nombre_producto           int64
longitud_descripcion_producto      int64
cantidad_fotos_producto            int64
peso_g_producto                    int64
longitud_cm_producto               int64
altura_cm_producto                 int64
ancho_cm_producto                  int64
dtype: object

tablas de frecuencia
para la soluci´´on del caso de suso es necesario identificas una posible hipótesis  para validar a través de las herramientas que se revisarán en las siguientes líneas

la tabla de frecuencia es una herramienta que permite ordenar los datos, de manera que se presentan numéricamente las caracteristicas de la distribución de un conjunto de datos  o muestra 

hipótesis: las categorías de celulares es la que más productos vende.
se queire validar ya que se desea lanzar una campaña de descuento para traer más lcientes a la empresa, sin embargo se debe conocer cuál va a ser la categoría en la cual se invierta más publicidad.


In [7]:
#tabla de frecuencia univariada
#se va a realizar una tabla para analizar las categorías que
#se venden en el comerciop

frec_datos=datos["nombre_categoria_producto"].value_counts().reset_index()
frec_datos.head(8)
frec_datos.dtypes

nombre_categoria_producto    object
count                         int64
dtype: object

In [8]:
#frecuencia relativa
frec_datos["fre_relative_categoria_prodcuto"]=frec_datos["count"].apply(lambda x: x/frec_datos["count"].sum())
frec_datos.head()

nombre_categoria_producto  count  fre_relative_categoria_prodcuto
0                Tecnología    543                         0.053582
1                  Deportes    537                         0.052990
2         Frutas y verduras    532                         0.052497
3                Carnicería    523                         0.051608
4           Ropa de adultos    521                         0.051411

In [9]:
#frecuencia absoluta y relatica acumulada con sumsum()
#absoluta acumulada
frec_datos["frec__acumu_abs_categoria"]=frec_datos["count"].cumsum()
#relatica acumulada
frec_datos["frec_acum_rela_categoria"]=frec_datos["fre_relative_categoria_prodcuto"].cumsum()
frec_datos.head(20)

nombre_categoria_producto  count  fre_relative_categoria_prodcuto  \
0                 Tecnología    543                         0.053582   
1                   Deportes    537                         0.052990   
2          Frutas y verduras    532                         0.052497   
3                 Carnicería    523                         0.051608   
4            Ropa de adultos    521                         0.051411   
5                 Ferretería    519                         0.051214   
6                   Juguetes    519                         0.051214   
7              Ropa infantil    512                         0.050523   
8          Electrodomésticos    510                         0.050326   
9                 Dormitorio    509                         0.050227   
10    Productos ecoamigables    501                         0.049438   
11                   Lácteos    501                         0.049438   
12                    Libros    499                         0.049240   
13           Tarjetas regalo    497                         0.049043   
14                  Mascotas    496                         0.048944   
15                     Salud    491                         0.048451   
16                 Celulares    488                         0.048155   
17                   Muebles    486                         0.047957   
18                     Licor    475                         0.046872   
19                    Bebés     475                         0.046872   

    frec__acumu_abs_categoria  frec_acum_rela_categoria  
0                         543                  0.053582  
1                        1080                  0.106572  
2                        1612                  0.159068  
3                        2135                  0.210677  
4                        2656                  0.262088  
5                        3175                  0.313302  
6                        3694                  0.364515  
7                        4206                  0.415038  
8                        4716                  0.465364  
9                        5225                  0.515591  
10                       5726                  0.565029  
11                       6227                  0.614466  
12                       6726                  0.663706  
13                       7223                  0.712749  
14                       7719                  0.761693  
15                       8210                  0.810144  
16                       8698                  0.858299  
17                       9184                  0.906256  
18                       9659                  0.953128  
19                      10134                  1.000000

análisis:la tabla de frecuencias nos muestra las siguiente información

1.frecuencia absoluta: númeor de veces que aparece un valor
2.frecuencia relativa:es el resultaod de dividir la frecunecia absoluta de un producto por el número total de datos
frecuencia acumulada absoluta: la suma de las frecunecias absolutas
frecuencoas acumulada relativa: el resultado de dividir la frecuencia acumulad apor el n´+umero total de informacion

In [10]:
#tabla de frecuencia bivariada
##tabla que muestra la frecuencia por ciudad y categoría del producto
pd.crosstab(index=datos["nombre_categoria_producto"],columns=datos["precio"])

precio                     0.01     0.03     0.04     0.05     0.08     \
nombre_categoria_producto                                                
Bebés                            2        0        1        0        0   
Carnicería                       0        0        0        0        0   
Celulares                        0        0        0        1        0   
Deportes                         0        0        1        0        0   
Dormitorio                       0        0        1        0        0   
Electrodomésticos                0        0        0        0        0   
Ferretería                       0        0        0        0        0   
Frutas y verduras                0        0        0        0        0   
Juguetes                         0        0        0        0        0   
Libros                           0        0        0        0        0   
Licor                            0        0        0        0        0   
Lácteos                          0        0        0        0        0   
Mascotas                         0        0        0        0        0   
Muebles                          0        0        0        0        0   
Productos ecoamigables           0        0        0        0        1   
Ropa de adultos                  0        0        0        0        0   
Ropa infantil                    0        0        0        0        0   
Salud                            0        1        0        0        0   
Tarjetas regalo                  0        0        0        0        0   
Tecnología                       0        0        0        0        0   

precio                     0.09     0.11     0.12     0.14     0.15     ...  \
nombre_categoria_producto                                               ...   
Bebés                            0        0        0        0        0  ...   
Carnicería                       0        0        0        0        0  ...   
Celulares                        0        0        1        0        0  ...   
Deportes                         0        0        0        0        0  ...   
Dormitorio                       1        0        0        1        0  ...   
Electrodomésticos                0        0        0        0        0  ...   
Ferretería                       0        0        0        0        0  ...   
Frutas y verduras                0        0        0        0        0  ...   
Juguetes                         0        0        0        0        0  ...   
Libros                           0        0        0        0        0  ...   
Licor                            0        0        0        0        0  ...   
Lácteos                          0        0        0        0        0  ...   
Mascotas                         0        0        0        0        0  ...   
Muebles                          0        1        0        0        0  ...   
Productos ecoamigables           0        0        0        0        0  ...   
Ropa de adultos                  0        0        0        0        0  ...   
Ropa infantil                    0        0        0        0        0  ...   
Salud                            0        0        0        0        0  ...   
Tarjetas regalo                  0        0        0        0        0  ...   
Tecnología                       0        0        0        0        1  ...   

precio                     807.78   808.05   820.55   824.64   847.88   \
nombre_categoria_producto                                                
Bebés                            0        0        0        0        0   
Carnicería                       0        0        0        0        0   
Celulares                        1        1        0        0        0   
Deportes                         0        0        0        0        0   
Dormitorio                       0        0        0        0        0   
Electrodomésticos                0        0        0        0        0   
Ferretería                       0      

análisis: es posible ver en la tabla la cantidad de compras que tiene las diferentes categorías, la que más tiene es tecnología con 532 ventas en total 

## tendencia central
la tendencia central significa medir el centro de distribucion de ubicacion de valores de un conjunto de datos. Da una idea del valor promedio de las viraiables en el conjunto de datos y también una indicación de cuán ampliamente se distribuyen los valores en el conjunto de datos.
1. media
2. mediana
3. moda

Hipótesis : los precio en el comercio son mayores a 500, ya que el estudio de marketing arrojó que los clientes más recurrentes compran mayormente productos más económicos( debajo de 500), por lo tanto es necesario validar si en promedio la compañis tiene precios más competitivos

In [11]:
# media o prmedio
#promedio de precio
datos["precio"].mean()

119.7923465561476

In [12]:
#mediana
#mediana de precio, valor central de la ddistribución de los precio
datos["precio"].median()

83.12

In [13]:
#moda
#mvalor más común en los precio
datos["precio"].mode()

0     3.92
1    11.39
2    11.79
3    15.80
4    16.42
5    25.45
6    34.43
7    36.96
8    56.79
Name: precio, dtype: float64

Análisis: Es posible analizar que el promedio es inferior a 500. ubicandose en 120, la mediana que es el dato central se ubica en 83,12 y existen varios valores comunes y todo debajo de 60 lo que nos muestra que la mayor cantidad de precios son bajos y permiten ser más atractivos para los consumidores finales que compran mayormente precios debajo de 500

## medidas de localizacion
los cuantiles y percentiles son punetos de corte que dividen el rango de una distribución de probabilidad en intervalos continuos con probabilidades iguales o que dividen las observaciones en una muestra de la misma forma

Así pudes representar un porcentaje de datos a travpes de los cuantiles para identificar como estanlocalizado

hipotesis: la mitad (50%) de los productos pesan más de 500 gramos. ya que la empresa de mensajería quiere realizar un descuento si los productos pesan menos de 500 gramos, así la compañía podría bajar los precios de envios a la mitad de los productos que se venden

In [14]:
#cuantiles
#calcular el 20%, 50 y 70% del peso de los productos
datos["peso_g_producto"]. quantile([0.2, 0.5,0.7])

0.2     470.0
0.5    1458.5
0.7    2535.1
Name: peso_g_producto, dtype: float64

Análisis: El percentil 50 tiene un valor de 1.4 kg, lo cual nos indica que la mitad de los productos pesan hasta 2.9 veces el valor de la hipótesis. esto nos rechaza la hipótesis y nos dice que no se puede realizar el dexscuento sobre el envio de lso productos, sin embargo deja la puerta abierta para encontrar la cantidad de prodcutos que puedan entrar ene el descuento en este caso dería el 20 %

## medida s de dispersión
se utiliza las estadisticas de dispersión par dar un número único que describa que tan compacto o disperso es un conjunto de observaciones.- estas se complementan de las medidas mencionadas anteriormente
1. desviación estandar
2. rango intercuartilico
3. rango
4. coeficiente de variación

Hipótesis: la variacion de las longitudes de los datos es mayor a 20cm. 
hay empresas de mensajería que son más económicas en envios , per estas tienen cubículos individuales de 20 cm. siendo el caso podría cambairse de empresa de emnsajería

In [15]:
##deviación estándar
#representa qeu tan lejos de la media se encuentra una observación
media=datos["longitud_cm_producto"].mean()
print(media)
datos["longitud_cm_producto"].std()

30.071936056838364


15.292866357696994

In [16]:
##rango intercuartílico
#es la diferencia entre el tercer y el primer cuartil
datos["longitud_cm_producto"].quantile(0.75)-datos["longitud_cm_producto"].quantile(0.25)

22.0

In [17]:
#rango
#representa la diferencia que existe entre el producto con una longitud más grande y el de la longitud más pequeña
datos["longitud_cm_producto"].max()-datos["longitud_cm_producto"].min()

91

In [18]:
#coeficiente de variación
#es la divición de la desviación estándar y la emdia
datos["longitud_cm_producto"].std()/datos["longitud_cm_producto"].mean()


0.5085427931474799

FUNCION EN PANDAS QUE HAC ETODO


In [19]:
datos.describe()

precio   valor_flete  codigo_postal_vendedor  \
count  10134.000000  10134.000000            10134.000000   
mean     119.792347     20.053246            49991.684626   
std      119.340785     20.106163            22463.895340   
min        0.010000      0.000000             5001.000000   
25%       34.155000      5.760000            50001.000000   
50%       83.120000     13.980000            52323.000000   
75%      165.515000     27.455000            54003.000000   
max     1262.940000    183.150000            99001.000000   

       longitud_nombre_producto  longitud_descripcion_producto  \
count              10134.000000                   10134.000000   
mean                  20.074798                      19.845372   
std                   11.721956                      11.854161   
min                    0.000000                       0.000000   
25%                   10.000000                      10.000000   
50%                   20.000000                      20.000000   
75%                   30.000000                      30.000000   
max                   40.000000                      40.000000   

       cantidad_fotos_producto  peso_g_producto  longitud_cm_producto  \
count             10134.000000     10134.000000          10134.000000   
mean                 20.110026      2106.109236             30.071936   
std                  11.802456      2093.734472             15.292866   
min                   0.000000         0.000000              0.000000   
25%                  10.000000       607.000000             19.000000   
50%                  20.000000      1458.500000             30.000000   
75%                  30.000000      2941.000000             41.000000   
max                  40.000000     20661.000000             91.000000   

       altura_cm_producto  ancho_cm_producto  
count        10134.000000       10134.000000  
mean            17.915433          22.905861  
std             11.360360          11.151559  
min              0.000000           0.000000  
25%              9.000000          15.000000  
50%             17.000000          23.000000  
75%             26.000000          30.000000  
max             64.000000          72.000000

acá se mestran las principales estadísticas descriptivas de resumen como:
1. count
2. mean
3. std
4. min and max
5. percentil 25, 50, 75